sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryAPITests.groovy

```diff
@@ -382,6 +382,42 @@ class DirectoryAPITests extends APISpec {
      */
     @Unroll
     def "List files and directories args"() {
+        given:
+        primaryDirectoryClient.create()
+        def nameList = new LinkedList()
+        def dirPrefix = namer.getRandomName(60)
+        for (int i = 0; i < 2; i++) {
+            def subDirClient = primaryDirectoryClient.getSubdirectoryClient(dirPrefix + i)
+            subDirClient.create()
+            for (int j = 0; j < 2; j++) {
+                def num = i * 2 + j + 3
+                subDirClient.createFile(dirPrefix + num, 1024)
+            }
+        }
+        primaryDirectoryClient.createFile(dirPrefix + 2, 1024)
+        for (int i = 0; i < 3; i++) {
+            nameList.add(dirPrefix + i)
+        }
+
+        when:
+        def fileRefIter = primaryDirectoryClient.listFilesAndDirectories(namer.getResourcePrefix() + extraPrefix, maxResults, null, null).iterator()
+
+        then:
+        for (int i = 0; i < numOfResults; i++) {
+            Objects.equals(nameList.pop(), fileRefIter.next().getName())
+        }
+        !fileRefIter.hasNext()
+
+        where:
+        extraPrefix   | maxResults | numOfResults
+        ""            | null       | 3
+        ""            | 1          | 3
+        "noOp"        | 3          | 0
+    }
```

Comment:
This is a reversion to the old test, keeping old and new service version functionality tested separately.